services/uploads/src/antivirus.js

```diff
@@ -33,25 +39,18 @@ async function isS3FileTooBig(s3ObjectKey, s3ObjectBucket) {
 }
 
 /**
- * Returns the /tmp local path from an s3ObjectKey
- * @param {string} s3ObjectKey Key of the s3 object
- * @return {string} Path of the s3 object on the local fs
- **/
-function pathFromObjectKey(s3ObjectKey) {
-    const downloadDir = `/tmp/download`;
-
-    // remove problematic characters from filename
-    let sanitizedFilename = s3ObjectKey.replace(/[^a-zA-Z0-9]/g, '');
-    return `${downloadDir}/${path.basename(sanitizedFilename)}`;
-}
-
+ * Download a file from S3 to a local temp directory.
+ * @param {string} s3ObjectKey       Key of S3 Object
+ * @param {string} s3ObjectBucket    Bucket of S3 object
+ * @return {Promise<string>}         Path to downloaded file
+ */
 function downloadFileFromS3(s3ObjectKey, s3ObjectBucket) {
-    const downloadDir = `/tmp/download`;
-    if (!fs.existsSync(downloadDir)) {
-        fs.mkdirSync(downloadDir);
+    if (!fs.existsSync(constants.TMP_DOWNLOAD_PATH)) {
+        fs.mkdirSync(constants.TMP_DOWNLOAD_PATH);
     }
 
-    let localPath = pathFromObjectKey(s3ObjectKey);
+    const tmpFileName = `${crypto.randomUUID()}.tmp`;
+    let localPath = `${constants.TMP_DOWNLOAD_PATH}${tmpFileName}`;
```

Comment:
ugh, i should have reviewed this in their PR. I wish this was actually shared code!! We should be using path.join here.

Reply:
yup good catch, I'll fix it